src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -290,8 +491,24 @@ inline void Details::DocumentStatisticsTrainingOnlyPolicy::fit(InputType const &
 }
 
 inline DocumentStatisticsAnnotationData Details::DocumentStatisticsTrainingOnlyPolicy::complete_training(void) {
-    IndexMap                                termIndex(Microsoft::Featurizer::Featurizers::Components::CreateIndexMap<std::string>(_termFrequency, IndexMap()));
-    return DocumentStatisticsAnnotationData(std::move(_termFrequency), std::move(termIndex), std::move(_totalNumDocuments));
+
+    //prune map by maxDf and minDf
+    FrequencyMap prunedTermFreqByDf = PruneTermFreqMap(_termFrequency, _minDf, _maxDf, static_cast<std::float_t>(_totalNumDocuments));
+
+    //prune map by maxFeatures
+    FrequencyMap prunedTermFreqByMaxFeatures = PruneTermFreqMap(prunedTermFreqByDf, _maxFeatures);
```

Comment:
As written, there will be 2 simultaneous copies of the frequency map - rewrite this so we only have 1 copy of the map.